### PR TITLE
Python: Add CMK support to create Index

### PR DESCRIPTION
**Pull request by charotAmine**
### Motivation and Context

Thank you for your contribution to the semantic-kernel repo!
Please help reviewers and future users, providing the following information:
  1. Why is this change required?
      When we try to create an index in AI Search where the encryption is enabled we have this issue : 
        Code: OperationNotAllowed
        Message: The creation of non-encrypted Indexes is not alowed when encryption policy is enforced.
      Which make the operation create_collection unusable
  2. What problem does it solve?
      Use create_collection function
  3. What scenario does it contribute to?
      If the customer uses the CMK for the encryption
  4. If it fixes an open issue, please link to the issue here.

### Description

The Azure Search Documents supports the encryption key with the class SearchResourceEncryptionKey, adding it (Free to the customer to use it or not) resolve the issue. 
Maybe it may be use it in save_information function as well ?

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py [hunk 1]
```diff
@@ -136,6 +141,7 @@ async def create_collection(
                 name=collection_name.lower(),
                 fields=get_index_schema(self._vector_size),
                 vector_search=vector_search,
+                encryption_key=search_resource_encryption_key,
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:144]: could you have a look to see if you can add a unit test for this, that tests both with and without key?

Comment (charotAmine) [reply python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:144]: Hello @eavanvalkenburg , I have mocked and added the tests.

File: python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py [hunk 2]
```diff
@@ -83,6 +84,7 @@ async def create_collection(
         self,
         collection_name: str,
         vector_config: Optional[HnswVectorSearchAlgorithmConfiguration] = None,
+        search_resource_encryption_key: SearchResourceEncryptionKey = None,
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:87]: Tiny comment: I'd love to see the Optional type hint here for type checking, to keep it consistent with the optional type hint for the parameter before it.

Comment (charotAmine) [reply python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py]: hHello @moonbox3  I have added the optional type hint !

**Review by moonbox3 — APPROVED** [review]
LGTM

**Review by eavanvalkenburg — APPROVED** [review]

**Pull request merged**